src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -419,7 +419,21 @@ private void FireProcessingQueueEmpty()
             Metrics.LastBlockProcessingTimeInMs = blockProcessingTimeInMicrosecs / 1000;
             Metrics.RecoveryQueueSize = _recoveryQueue.Count;
             Metrics.ProcessingQueueSize = _blockQueue.Count;
-            _stats.UpdateStats(lastProcessed, blockProcessingTimeInMicrosecs);
+
+            Transaction[] txs = suggestedBlock.Transactions;
+            Address beneficiary = suggestedBlock.Header.GasBeneficiary;
+            Transaction lastTx = txs.Length > 0 ? txs[^1] : null;
+            bool isMev = false;
+            if (lastTx is not null && (lastTx.SenderAddress == beneficiary || _alternateMevPayees.Contains(lastTx.SenderAddress)))
+            {
+                // Mev reward with in last tx
+                beneficiary = lastTx.To;
+                isMev = true;
+            }
+            UInt256 beforeBalance = _stateReader.GetBalance(processingBranch.Root, beneficiary);
+            UInt256 afterBalance = _stateReader.GetBalance(suggestedBlock.StateRoot, beneficiary);
+            UInt256 rewards = beforeBalance < afterBalance ? afterBalance - beforeBalance : default;
```

Comment:
- Should all of this be in `BlockchainProcessor`, sounds like it should be more in the `UpdateStats`?
- As it is doing additional state reads, maybe it should be in background thread?

Reply:
Moved to UpdateStats background thread